### PR TITLE
Migrate Docker example DAGs to new design 

### LINKED ISSUE #22444: Migrate Docker example DAGs to new design
There is a new design of system tests that was introduced by the [AIP-47](https://cwiki.apache.org/confluence/display/AIRFLOW/AIP-47+New+design+of+Airflow+System+Tests).

All current example dags need to be migrated and converted into system tests, so they can be run in the CI process automatically before releases.

This is an aggregated issue for all example DAGs related to `Docker` provider. It is created to track progress of their migration.

List of paths to example DAGs:
- [x] airflow/providers/docker/example_dags/example_docker_swarm.py
- [x] airflow/providers/docker/example_dags/example_docker.py
- [x] airflow/providers/docker/example_dags/example_docker_copy_data.py

### PR DESCRIPTION
**Pull request by Bowrna**
closes: #22444

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragement file, named `{pr_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: tests/system/providers/docker/example_docker.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    # t1 >> t3
+    # t3 >> t4
+
+    (
+        # TEST BODY
+        t1
+        >> [t2, t3]
+        >> t4
+    )
+
```
Comment (Bowrna) [inline tests/system/providers/docker/example_docker.py:59]: Is the task present inside above parenthesis is the right representation of 
    # t1 >> t2
    # t1 >> t3
    # t3 >> t4 ?

If not can you tell me how to represent it?
cc: @potiuk @bhirsz @mnojek

**Comment by potiuk** [comment]
I think you should run `breeze self-update` :) The images generated are from an old rich version and by self-upgrade you will get the latest one.

Then what you need to do to regenerate them - delete `images/breeze/output-commands-hash.txt` and run `breeze static-checks -t update-breeze-files`. 

BTW. I will add a separate command to regnerate those images a bit more easily.

**Comment by potiuk** [comment]
The "regenerate-breeze-command" command is added here: https://github.com/apache/airflow/pull/24216

**Comment by Bowrna** [comment]
@potiuk I made the changes you suggested and now I don't see the command images in the file diff. Thanks :)

**Review by potiuk — APPROVED** [review]

**Pull request merged**